### PR TITLE
[docs] Migrate Expo Router docs form home to Guides

### PR DESCRIPTION
**Pull request by amandeepmittal**
# Why

Closes ENG-10570

# How

This PR:
- Moves all docs from Home > Routing section to Expo Router
- Deletes "Next steps" doc under Routing
- Adds client and server side redirects as the URL of these migrated pages have been modified to follow the convention of using `router` instead of `routing`
- Adds a link to Expo Router section from Home > User Interface > Next steps
- Update all internal links that were referencing to `/routing/` before, to use `/router/`

# Test Plan

/router/introduction/

### Preview

**Expo Router sidebar section**
CleanShot 2023-11-29 at 22 48 05@2x

**User interface > Next steps page containing a `BoxLink` to Expo Router group**

CleanShot 2023-11-29 at 22 47 55@2x

# Checklist

- [ ] Documentation is up to date to reflect these changes (eg: https://docs.expo.dev and README.md).
- [ ] Conforms with the [Documentation Writing Style Guide](https://github.com/expo/expo/blob/main/guides/Expo%20Documentation%20Writing%20Style%20Guide.md)
- [ ] This diff will work correctly for `npx expo prebuild` & EAS Build (eg: updated a module plugin).

### CODE REVIEW

File: docs/pages/guides/progressive-web-apps.mdx [hunk 1]
```diff
@@ -7,7 +7,7 @@ import { CODE } from '~/ui/components/Text';
...
+> This guide is oriented towards apps using webpack. If you use [Metro for web](/guides/customizing-metro/#web-support) (such as with [Expo Router](/router/introduction)), you can still build a PWA. However, the Expo tooling suggested in this guide is not all intended for it. Use the [`web.output`](/versions/latest/config/app/#output) property with `single` as its default value to generate a SPA. You can then modify your **public/index.html** to add meta tags and PWA assets, and use [`expo-router/head`](/router/reference/static-rendering/#meta-tags) to set meta tags on each route. [Workbox](https://web.dev/learn/pwa/workbox/) is recommended for service workers.
```
Comment (keith-kurak) [inline docs/pages/guides/progressive-web-apps.mdx:10]:
```suggestion
> This guide is oriented towards apps using Webpack. If you use [Metro for web](/guides/customizing-metro/#web-support) (such as with [Expo Router](/router/introduction)), you can still build a PWA. However, the Expo tooling suggested in this guide is not intended for it. Use the [`web.output`](/versions/latest/config/app/#output) property with `single` as its default value to generate a SPA. You can then modify your **public/index.html** to add meta tags and PWA assets, and use [`expo-router/head`](/router/reference/static-rendering/#meta-tags) to set meta tags on each route. [Workbox](https://web.dev/learn/pwa/workbox/) is recommended for service workers.
```

File: docs/pages/router/reference/authentication.mdx [hunk 2]
```diff
@@ -182,7 +182,7 @@ export default function Root() {
 
 <Step label="3">
 
-Create a nested [Layout route](/routing/layouts) that checks whether users are authenticated before rendering the child route components. This layout route redirects users to the sign-in screen if they are not authenticated.
+Create a nested [Layout route](/router/layouts) that checks whether users are authenticated before rendering the child route components. This layout route redirects users to the sign-in screen if they are not authenticated.
```
Comment (keith-kurak) [inline docs/pages/router/reference/authentication.mdx:185]:
```suggestion
Create a nested [layout route](/router/layouts) that checks whether users are authenticated before rendering the child route components. This layout route redirects users to the sign-in screen if they are not authenticated.
```

File: docs/pages/develop/project-structure.mdx [hunk 3]
```diff
@@ -9,7 +9,7 @@ When you create a new Expo project, a few files and folders are provided by defa
 
 The **App.js** file is the default screen of your project. It is the root file that you'll load after running the development server with `npx expo start`. You can edit this file to see the project update instantly. Generally, this file will contain root-level React Contexts and navigation.
 
-> [Expo Router](/routing/introduction/) is now available and supports file-based navigation in projects. With Expo Router, the route file will become **index.js** and all the screens in the project will be in the **app** folder.
+> [Expo Router](/router/introduction/) is now available and supports file-based navigation in projects. With Expo Router, the route file will become **index.js** and all the screens in the project will be in the **app** folder.
```
Comment (keith-kurak) [inline docs/pages/develop/project-structure.mdx:12]: I know this was already there, but I'm not sure what "the route file" is here. Is it the initial screen?

File: docs/pages/develop/user-interface/next-steps.mdx [hunk 4]
```diff
@@ -1,12 +1,20 @@
...
 ---
 
 import { BoxLink } from '~/ui/components/BoxLink';
+import { RouterLogo } from '@expo/styleguide';
 import { Brush01Icon, Settings02Icon, TableIcon } from '@expo/styleguide-icons';
 
+<BoxLink
+  title="Expo Router"
+  Icon={RouterLogo}
+  description="Expo Router is an open-source routing library for Universal React Native applications built with Expo."
```
Comment (keith-kurak) [inline docs/pages/develop/user-interface/next-steps.mdx:14]: I like getting the "file system" part in there somewhere, see the description in https://www.youtube.com/watch?v=608r8etX_cg&t=678s
```suggestion
  description="Expo Router is an open-source file system-based routing library for Universal React Native applications built with Expo."
```

File: docs/pages/router/migrate/from-react-navigation.mdx [hunk 5]
```diff
@@ -84,7 +84,7 @@ export default function App() {
 
 Create an **app** directory at the root of your repo, or in a root **src** directory.
 
-Layout the structure of your app by creating files according to the [creating pages guide](/routing/create-pages/). Best practice dictates that you use kebab-case and lowercase letters for filenames.
+Layout the structure of your app by creating files according to the [creating pages guide](/router/create-pages/). Best practice dictates that you use kebab-case and lowercase letters for filenames.
```
Comment (keith-kurak) [inline docs/pages/router/migrate/from-react-navigation.mdx:87]:
```suggestion
Layout the structure of your app by creating files according to the [creating pages guide](/router/create-pages/). Kebab-case and lowercase letters are considered best practice for route filenames.
```

**Review by keith-kurak — APPROVED** [review]
LGTM!

Added some take-it-or-leave-it nits for parts that showed up in the diff that weren't really part of the scope of the change.

File: docs/pages/develop/project-structure.mdx [hunk 6]
```diff
@@ -9,7 +9,7 @@ When you create a new Expo project, a few files and folders are provided by defa
 
 The **App.js** file is the default screen of your project. It is the root file that you'll load after running the development server with `npx expo start`. You can edit this file to see the project update instantly. Generally, this file will contain root-level React Contexts and navigation.
```
Comment (amandeepmittal) [inline docs/pages/develop/project-structure.mdx:10]:
```suggestion
The **App.js** file is the default screen of your project. It is the root file that loads after running the development server with `npx expo start`. You can edit this file to see the project update instantly. Generally, this file will contain root-level React Contexts and navigation.
```

**Pull request merged**